clarity/src/vm/functions/database.rs

```diff
@@ -924,3 +947,190 @@ pub fn special_get_burn_block_info(
         }
     }
 }
+
+/// Handles the `get-stacks-block-info?` special function.
+/// Interprets `args` as variables `[property-name, block-height]`, and returns
+/// a property value determined by `property-name`:
+/// - `id-header-hash` returns the index block hash at `block-height`
+/// - `header-hash` returns the header hash at `block-height`
+/// - `time` returns the block time at `block-height`
+///
+/// # Errors:
+/// - CheckErrors::IncorrectArgumentCount if there aren't 2 arguments.
+/// - CheckErrors::GetStacksBlockInfoExpectPropertyName if `args[0]` isn't a ClarityName.
+/// - CheckErrors::NoSuchStacksBlockInfoProperty if `args[0]` isn't a StacksBlockInfoProperty.
+/// - CheckErrors::TypeValueError if `args[1]` isn't a `uint`.
+pub fn special_get_stacks_block_info(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    // (get-stacks-block-info? property-name block-height-uint)
+    runtime_cost(ClarityCostFunction::BlockInfo, env, 0)?;
+
+    check_argument_count(2, args)?;
+
+    // Handle the block property name input arg.
+    let property_name = args[0]
+        .match_atom()
+        .ok_or(CheckErrors::GetStacksBlockInfoExpectPropertyName)?;
+
+    let block_info_prop = StacksBlockInfoProperty::lookup_by_name(property_name).ok_or(
+        CheckErrors::NoSuchStacksBlockInfoProperty(property_name.to_string()),
+    )?;
+
+    // Handle the block-height input arg.
+    let height_eval = eval(&args[1], env, context)?;
+    let height_value = match height_eval {
+        Value::UInt(result) => Ok(result),
+        x => Err(CheckErrors::TypeValueError(TypeSignature::UIntType, x)),
+    }?;
+
+    let height_value = match u32::try_from(height_value) {
+        Ok(result) => result,
+        _ => return Ok(Value::none()),
+    };
```

Comment:
```suggestion
    let Ok(height_value) = u32::try_from(height_value) else {
        return Ok(Value::none())
    };
```

Reply:
Fixed in 1fad8d226e8081b4204c6a6ecc44c5ff8038013b